src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -554,15 +554,37 @@ public function nodeName()
     }
 
     /**
-     * Returns the node value of the first node of the list.
+     * Returns the text of the first node of the list.
+     *
+     * Pass true as the 2nd argument to normalize whitespaces.
      *
      * @param mixed $default When provided and the current node is empty, this value is returned and no exception is thrown
      *
      * @return string The node value
      *
      * @throws \InvalidArgumentException When current node is empty
      */
-    public function text(/* $default = null */)
+    public function text(/* $default = null, $normalizedWhitespace = true */)
+    {
+        if (\func_num_args() > 1) {
+            if (\is_bool(func_get_arg(1))) {
+                return trim(preg_replace('/\s++/', ' ', $this->textContent(func_get_arg(0))));
+            }
+
+            return $this->textContent(func_get_arg(0));
+        }
+
+        return $this->textContent();
+    }
+
+    /**
+     * Returns the node value of the first node of the list.
+     *
+     * @return string The node value
+     *
+     * @throws \InvalidArgumentException When current node is empty
+     */
+    public function textContent(/* $default = null */)
```

Comment:
Why do we need a separate method? This might be confusing to me.

Reply:
cc @xabbuh do we still need a new method ?

Reply:
I don't remember the state of the discussion in the previous PR. IIRC some people preferred an additional argument for `text()`, others wanted a new method. We should make a decision on which solution we want first I think.

Reply:
I guess we can close this and make the decision in the issue before reopening or doing something ?

Reply:
I think we should:

1. Add a boolean optional argument to the existing text() method (I know, boolean params are bad in general ... but in this case it seems the best option ... or the least bad option).
2. Name that argument as `$removeWhitespaces` (false by default to keep BC) (instead of the proposed `$normalizedWhitespace` name)
3. Do more than `trim()`. To be really useful, it must do what's originally proposed in #21302:

```php
$text = $element->text();
$text = preg_replace("/(\n|\t)/", '', $text);
$text = preg_replace("/ {2,}/", ' ', $text);
```

Reply:
please remove the textContent method